lib/turn.rb

```diff
@@ -0,0 +1,84 @@
+require "./lib/card.rb"
+require "./lib/deck.rb"
+require "./lib/player.rb"
+
+class Turn 
+  attr_reader :player1,
+              :player2,
+              :spoils_of_war 
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+  end
+
+  def type 
```

Comment:
A few things are happening in this method that could be cleaned up. A nested `if` statement is sometimes necessary, however, if we can avoid it, that's even better. Nested `if` statements are prone to bugs and are hard to test. 

Another thing I notice that that there are two scenarios each that result in a type of `:war` and `:basic`. Ideally, our logic would have one path for every outcome (sometimes the ideal is not possible but is preferred when we can).

Something like this would accomplish the same thing but with fewer levels of logic:
```ruby
  def type
    if @player1.deck.rank_of_card_at(0) != @player2.deck.rank_of_card_at(0)
      :basic
    elsif (@player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)) && (@player1.deck.rank_of_card_at(2) == @player2.deck.rank_of_card_at(2)) && (player1.deck.cards.length >= 3 && player2.deck.cards.length >= 3)
      :mutually_assured_destruction
    elsif @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
      :war
    end
  end
```

I would also probably create some helper methods though to improve readability. One thing I noticed is that I played around with this logic a few different ways and your tests passed every time, even though I was missing the `cards.length >= 3` portion for a while. This tells me there could be more thorough testing. Tests should have failed when I forgot to include that piece of logic. 

A helper method would be something like:
```ruby
  def same_rank_at_zero?
    @player1.deck.rank_of_card_at(0) == @player2.deck.rank_of_card_at(0)
  end
```

And then we can test that logic separately as well as call the `same_rank_at_zero?` method in our `if` statement instead of typing out that same comparison logic every time we need to check (currently happening on lines 18, 20, 26).